hoomd/md/manifold.py

```diff
@@ -0,0 +1,367 @@
+# Copyright (c) 2009-2019 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+# Maintainer: joaander / All Developers are free to add commands for new features
+
+import hoomd
+from hoomd.md import _md
+from hoomd import _hoomd
+from hoomd.operation import _HOOMDBaseObject
+from hoomd.data.parameterdicts import ParameterDict, TypeParameterDict
+from hoomd.data.typeconverter import OnlyIf, to_type_converter
+from collections.abc import Sequence
+
+# A manifold in hoomd reflects a Manifold in c++. It is responsible to define the manifold 
+# used for RATTLE integrators and the active force constraints.
+class Manifold(_HOOMDBaseObject):
+    r"""Base class manifold object.
+
+    Manifold defines a positional constraint to a given set of particles. A manifold can 
+    be applied to a RATTLE method and/or the active force class. The degrees of freedom removed from 
+    the system by constraints are correctly taken into account, i.e. when computing temperature 
+    for thermostatting and/or logging.
+
+    All manifolds are described by implicit functions.
+
+    Note:
+        Users should not instantiate :py:class:`Manifold` directly, but should instead instantiate
+        one of its subclasses defining a specific manifold geometry.
+
+    Warning:
+        Only one manifold can be applied to the methods/active forces.
+
+    """
+    def _attach(self):
+        self._apply_param_dict()
+        self._apply_typeparam_dict(self._cpp_obj, self._simulation)
+
+    @staticmethod
+    def _preprocess_unitcell(value):
+        if isinstance(value, Sequence):
+            if len(value) != 3:
+                raise ValueError(
+                    "Expected a single int or a sequence of three ints.")
+            return tuple(value)
+        else:
+            return (value,value,value)
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return all(getattr(self, attr) == getattr(other, attr) for attr in self._param_dict)
+
+    def _setattr_param(self, attr, value):
+        raise AttributeError("Manilfolds are immutable and {} cannot be set "
+                            "after initialization".format(attr))
+
+
+class Cylinder(Manifold):
+    R""" Cylinder manifold.
+
+    Args:
+        r (`float`): radius of the cylinder constraint (in distance units).
+        P (`tuple`[ `float`, `float`, `float`]): point defining position of the cylinder axis (default origin).
```

Comment:
```suggestion
        P (`tuple` [`float`, `float`, `float`]): point defining position of the cylinder axis (default origin).
```
Sphinx is not formatting these correctly. I think adding this space will fix it. There are several places in `methods.py` that need this fix. Check the rendered sphinx docs to make sure it looks right.